text/0000-rustdoc-json.md

```diff
@@ -256,9 +256,10 @@ Name      | Type    | Description
 `name`    | String  | The name of the Item, if present. Some Items, like impl blocks, do not have names.
 `span`    | [Span](#Span) | (*Optional*) The source location of this Item.
 `visibility` | String | `"default"`, `"public"`, `"crate"`, or `"restricted"` (`pub(path)`).
-`restricted_path` | String | (*Optional*) If `visitility == "restricted"`, this field contains the path that it's restricted to.
+`restricted_path` | String | (*Optional*) If `visibility == "restricted"`, this field contains the path that it's restricted to.
 `docs`    | String  | The extracted documentation text from the Item.
-`attrs`   | [String] | The attributes (other than doc comments) on the Item, rendered as strings.
+`links`   | [[ID](#ID)] | A list of items corresponding to any intra-doc links in `docs` in order of appearance.
```

Comment:
What happens if there are broken links? For the following code, I think rustdoc will internally store `[<Def Id of f>]`, so you'd have no way to know whether that should belong to `f` or to `MissingItem`.

```rust
/// Link to [`MissingItem`], [`f`]
pub fn f() {}
```

I think a map from the name to the id is more resilient. Alternatively we could store `null` to mean an unresolved link, but that requires the consumer of the output to do some more work on their end.

Reply:
Yeah a mapping does sound better, i was worried at first that we couldn't use a map because of duplicate keys but that isn't an issue because there's no way to have the same name resolve to different stuff.

Reply:
Hmm now that you say that I'm not actually sure it's true ... consider:

```rust
pub struct S;
/// Link to [S]
pub fn f() {}
pub mod inner {
    pub struct S;
    /// Link to [S]
    pub fn g();
}
```
Maybe we should include the ID of the item being documented as one of the keys?

Reply:
I think we're still fine, this mapping is just for a single item's docstring so in that context it could only refer to one of them.